build/xtask/src/dist.rs

```diff
@@ -2138,6 +2156,95 @@ fn write_srec(
     Ok(())
 }
 
+fn write_elf(
+    sections: &BTreeMap<AbiSize, LoadSegment>,
+    kentry: AbiSize,
+    out: &Path,
+) -> Result<()> {
+    use goblin::container::{Container, Ctx, Endian};
+    use goblin::elf::header::{Header, EM_RISCV, ET_EXEC};
+    use goblin::elf::program_header::{PF_R, PF_W, PF_X, PT_LOAD};
+    use goblin::elf::ProgramHeader;
+    use scroll::Pwrite;
+
+    let ctx = Ctx::new(Container::Big, Endian::Little);
+
+    // Generate all the program headers and collect all the sections together.
+    let mut program_headers = Vec::new();
+    let mut sections_data = Vec::new();
+    for (&base, sec) in sections {
+        if sec.data.is_empty() {
+            continue;
+        }
+
+        // Incredibly simple algorithm that attempts to align where this program section appears in
+        // the file (what the loop adjusts) with its location in virtual address space (which was
+        // fixed by the compiler already).
+        while ((base as usize) & 0xfff) != (sections_data.len() & 0xfff) {
+            sections_data.push(0);
```

Comment:
Should it be filling with 0x0 or 0xff?

Reply:
I think the 'sections' only contain loadable segments? I'm not sure the padding value matters in practice aside from on-flash usage (which I thin is why you're asking). Whichever value we choose we should note in commentary there was thought put into the reason for the value chosen (and the alternatives).

Reply:
Yes, I was thinking about the flash usage. If we are adding some kind of padding to align the sections, then having `0xff` just makes it easier to skip the write on flash since it erases default to 0xff.

I agree that we should add comments around the reason behind some of the choices so that it is helpful later on.

Reply:
Adjusted to fill with 0xFF, added a comment along @furquan-rivos's reasoning.